src/parsers/manifestjson.js

```diff
@@ -70,6 +70,9 @@ export default class ManifestJSONParser extends JSONParser {
       };
     } else {
       // We've parsed the JSON; now we can validate the manifest.
+      if (this.parsedJSON.browser_specific_settings) {
```

Comment:
Besides overriding the content of `applications` with `browser_specific_settings`, I'm thinking that we should also emit a linting error if both `applications.gecko` and `browser_specific_settings.gecko` exists in the same extension manifest.

@eviljeff (based on what we are actually doing on the AMO server side) what do you think? do you see any reason to don't block the submissions of extensions with both the manifest properties?

Reply:
It would be best to follow Firefox' behavior as close as possible. It would be a bad experience for developers if they successfully tested their add-on in Firefox (`about-debugging` or `web-ext`) and then are presented with an error when they try to submit to AMO.

Reply:
> @wagnerand there are already some cases where the extension install successfully on Firefox but linting errors are reported when checked with the addons-linter, e.g. if the manifest contains a content script which is missing from the resources actually included in the extension package.

Can you file a bug about that, please? The linter shouldn't behave differently than Firefox.

I see your point about potentially surprising behavior, but I think using a warning (and not an error) would be best here.

Reply:
> I see your point about potentially surprising behavior, but I think using a warning (and not an error) would be best here.

Sounds good to me, if a linting warning is the behavior that we agree on, then I'm also ok with it:
it matches Firefox behavior and the linter will not be totally silent if it met an extension with this kind of potential issue.

> > @wagnerand there are already some cases where the extension install successfully on Firefox but linting errors are reported when checked with the addons-linter, e.g. if the manifest contains a content script which is missing from the resources actually included in the extension package.
> 
> Can you file a bug about that, please? The linter shouldn't behave differently than Firefox.

Filing a new issue to discuss and evaluate the removal of that linting rule sounds good to me, but to be honest I'm not sure that we should remove it completely, one of the reasons why we have that linting rule is that in the past some addon developers were confused about "why suddenly their extension's content script wasn't working anymore", and it was often due to issues with the file names (e.g. a file got renamed and it wasn't updated in the manifest to reflect the new file name).

I agree that the addons-linter should behave as similar as possible to Firefox, but in some cases Firefox is less strict about "which issues should prevent the extension from install successfully" to make it easier to test and investigate these issues or for "performance" reasons/concerns (e.g. when checking it explicitly at runtime in Firefox may be too expensive). 

On the other hand the role of the addons-linter (at least in my own opinion) is to provide some additional guarantees that the extension is going to work as it should for the user that installs it from AMO (at least in the cases where it is actually possible in practice).

Reply:
I agree with what you are saying and I believe warnings would be the right level for those things, where we do more checks than Firefox. We don't need to remove such rules entirely.

Reply:
Can you file a follow-up issue for adding a warning for having both properties? - I've not got capacity to keep working on this to add something extra right now. If you think it needs to be landed as a single patch I'll leave this issue for someone else to finish.

Reply:
@eviljeff sure thing, I'll deal with the follow up, I just filed it as #2294.